app/models/host/managed.rb

```diff
@@ -741,6 +741,13 @@ def root_pass
     Setting[:root_pass]
   end
 
+  def root_pass_source
+    return :host if self[:root_pass].present?
+    return :hostgroup if hostgroup.try(:root_pass).present?
+    return :global if Setting[:root_pass].present?
```

Comment:
These should probably be translated strings, e.g.
```suggestion
    return N_("host") if self[:root_pass].present?
    return N_("hostgroup") if hostgroup.try(:root_pass).present?
    return N_("global setting") if Setting[:root_pass].present?
```